Shared/Samples/Filter by definition expression or display filter/FilterByDefinitionExpressionOrDisplayFilterView.swift

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct FilterByDefinitionExpressionOrDisplayFilterView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The draw status of the map view.
+    @State private var drawStatus: DrawStatus?
+    
+    /// The count of features in the current extent of the map view.
+    @State private var featureCount = 0
+    
+    /// The filtering mode selected in the picker.
+    @State private var selectedFilterMode: FilterMode = .none
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        GeometryReader { geometryProxy in
+            MapViewReader { mapViewProxy in
+                MapView(map: model.map)
+                    .onDrawStatusChanged { drawStatus = $0 }
+                    .task(id: drawStatus) {
+                        // Updates the feature count when the map view finishes drawing.
+                        guard drawStatus == .completed else { return }
+                        
+                        // Creates an envelope from the frame of the map view.
+                        let viewRect = geometryProxy.frame(in: .local)
+                        guard let viewExtent = mapViewProxy.envelope(
+                            fromViewRect: viewRect
+                        ) else { return }
+                        
+                        do {
+                            // Gets the feature count contained within the envelope.
+                            featureCount = try await model.queryFeatureCount(extent: viewExtent)
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .errorAlert(presentingError: $error)
+            }
+        }
+        .overlay(alignment: .top) {
+            Text("Feature count: \(featureCount)")
```

Comment:
Consider something a bit simpler:

```suggestion
            Text("\(featureCount) feature(s)")
```